src/test/java/teammates/test/cases/webapi/DeleteFeedbackSessionActionTest.java

```diff
@@ -23,13 +28,52 @@ protected String getRequestMethod() {
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("Typical case");
+
+        CourseAttributes course = typicalBundle.courses.get("typicalCourse1");
+        FeedbackSessionAttributes session = typicalBundle.feedbackSessions.get("session1InCourse1");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+        };
+
+        DeleteFeedbackSessionAction deleteFeedbackSessionAction = getAction(params);
+
+        // assert it exists before deletion
+        assertEquals(session.getFeedbackSessionName(),
+                logic.getFeedbackSessionDetails(session.getFeedbackSessionName(),
+                        course.getId()).feedbackSession.getFeedbackSessionName());
+
+        logic.moveFeedbackSessionToRecycleBin(session.getFeedbackSessionName(), course.getId());
+
+        JsonResult result = getJsonResult(deleteFeedbackSessionAction);
+        MessageOutput messageOutput = (MessageOutput) result.getOutput();
+
+        assertEquals(messageOutput.getMessage(), "The feedback session is deleted.");
+        assertThrows(EntityDoesNotExistException.class, () -> {
```

Comment:
Instead of checking for an exception when you attempt to get the feedback session, just verify if logic returns null when you retrieve it: `assertNull(logic.getFeedbackSessionFromRecycleBin(session.getFeedbackSessionName(), course.getId());`

Reply:
Updated